app/controllers/insights_cloud/api/machine_telemetries_controller.rb

```diff
@@ -69,5 +70,14 @@ def ensure_branch_id
       @branch_id = cp_owner_id(@organization)
       return render_message "Branch ID not found for organization #{@organization.title}", :status => 400 unless @branch_id
     end
+
+    def update_host_insights_status
+      return unless request.path == '/redhat_access/r/insights/platform/ingress/v1/upload' ||
+                    request.path.include?('/redhat_access/r/insights/uploads/')
+
+      host_status = @host.get_status(InsightsClientReportStatus)
+      host_status.update(reported_at: DateTime.now)
```

Comment:
Instead of updating the database twice, you can update the in-memory record and then save it.